### PR TITLE
Cache has_release_permissions for a minute.

### PR DESCRIPTION
**Pull request by JTCunning**

### CODE REVIEW

File: src/sentry/api/bases/organization.py [hunk 1]
```diff
@@ -294,6 +297,12 @@ def has_release_permission(self, request, organization, release):
         Does the given request have permission to access this release, based
         on the projects to which the release is attached?
         """
-        return ReleaseProject.objects.filter(
-            release=release, project__in=self.get_projects(request, organization)
-        ).exists()
+        key = "release_perms:1:%s" % hash_values([request.user.id, organization.id, release.id])
```
Comment (mattrobenolt) [inline src/sentry/api/bases/organization.py]: The only thing I’m not sure about that I’d like verified, how does this work with API keys? request isn’t always a user.

Comment (JTCunning) [reply src/sentry/api/bases/organization.py]: Good point. I'll look at the get_projects further to see how it handles it

Comment (JTCunning) [reply src/sentry/api/bases/organization.py]: We'll need a way to coalesce `request.user.id` and `ApiKey.id` down to just get a unique identifier of the authn mechanism.

**Review by mattrobenolt — APPROVED** [review]
I don't think this will error.

🥉

**Pull request merged**